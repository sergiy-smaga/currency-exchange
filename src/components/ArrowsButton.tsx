import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import CompareArrowsIcon from '@mui/icons-material/CompareArrows';
import { useCurrencyContext } from '../context/CurrencyContext';

export const ArrowsButton = () => {
  const { fromCurrency, setFromCurrency, toCurrency, setToCurrency } =
    useCurrencyContext();

  const handleSwitch = () => {
    setFromCurrency(toCurrency);
    setToCurrency(fromCurrency);
  };

  return (
    <Grid item xs={2}>
      <Button
        onClick={handleSwitch}
        sx={{
          borderRadius: 1,
          height: '100%',
        }}
      >
        <CompareArrowsIcon sx={{ fontSize: 30 }} />
      </Button>
    </Grid>
  );
};
