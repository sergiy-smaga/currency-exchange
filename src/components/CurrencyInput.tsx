import Grid from '@mui/material/Grid';
import TextField from '@mui/material/TextField';
import { useCurrencyContext } from '../context/CurrencyContext';

export const CurrencyInput = () => {
  const { firstAmount, setFirstAmount } = useCurrencyContext();
  return (
    <Grid item xs={3}>
      <TextField
        value={firstAmount}
        onChange={(e) => setFirstAmount(e.target.value)}
        label="Change"
        fullWidth
        InputProps={{
          type: 'number',
        }}
      />
    </Grid>
  );
};
