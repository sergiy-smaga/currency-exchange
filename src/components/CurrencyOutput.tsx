import { useEffect, useState } from 'react';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import { useCurrencyContext } from '../context/CurrencyContext';

export const CurrencyOutput = () => {
  const { firstAmount, fromCurrency, toCurrency, rates } = useCurrencyContext();
  const [value, setValue] = useState<number>(0);

  useEffect(() => {
    const rateBuy = rates.find(
      (item) =>
        item.currencyCodeA === fromCurrency && item.currencyCodeB === toCurrency
    );
    const rateSell = rates.find(
      (item) =>
        item.currencyCodeA === toCurrency && item.currencyCodeB === fromCurrency
    );
    rateBuy
      ? setValue(Number(firstAmount) * rateBuy.rateBuy)
      : rateSell
      ? setValue(Number(firstAmount) / rateSell.rateSell)
      : setValue(Number(firstAmount)); //case of equal currencies
  }, [firstAmount, fromCurrency, rates, toCurrency]);

  return (
    <Grid item xs={3}>
      <TextField
        value={value.toFixed(2)}
        label="Get"
        fullWidth
        InputProps={{
          type: 'number',
          readOnly: true,
        }}
      />
    </Grid>
  );
};
