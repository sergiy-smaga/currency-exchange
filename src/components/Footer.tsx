import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';

export const Footer = () => {
  return (
    <Box sx={{ height: 100 }}>
      <Paper
        sx={{
          textAlign: 'center',
          height: '100%',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
        }}
      >
        2023 All rights reserved
      </Paper>
    </Box>
  );
};
