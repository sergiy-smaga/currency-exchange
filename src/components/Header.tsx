import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';

export const Header = () => {
  return (
    <Box sx={{ height: 100 }}>
      <Paper
        sx={{
          height: '100%',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
        }}
      >
        Header Text Logo
      </Paper>
    </Box>
  );
};
