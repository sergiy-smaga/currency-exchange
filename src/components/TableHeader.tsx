import { Grid } from '@mui/material';
import { styled } from '@mui/material/styles';
import Paper from '@mui/material/Paper';

const Item = styled(Paper)(({ theme }) => ({
  backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
  ...theme.typography.body2,
  padding: theme.spacing(1),
  textAlign: 'center',
  color: theme.palette.text.secondary,
}));

export const TableHeader = () => {
  return (
    <>
      <Grid item xs={4}>
        <Item>Currency/Current Date</Item>
      </Grid>
      <Grid item xs={4}>
        <Item>Buy</Item>
      </Grid>
      <Grid item xs={4}>
        <Item>Sell</Item>
      </Grid>
    </>
  );
};
