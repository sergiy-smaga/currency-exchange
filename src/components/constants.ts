export const CURRENCIES = [
  {
    code: 980,
    name: 'UAH',
  },
  {
    code: 840,
    name: 'USD',
  },
  {
    code: 978,
    name: 'EUR',
  },
];
