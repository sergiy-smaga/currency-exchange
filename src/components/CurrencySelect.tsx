import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import Grid from '@mui/material/Grid';
import MenuItem from '@mui/material/MenuItem';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import { CURRENCIES } from './constants';
import { SelectProps } from '../types';

export const CurrencySelect: React.FC<SelectProps> = ({ value, setValue }) => {
  const handleChange = (event: SelectChangeEvent) => {
    setValue(Number(event.target.value));
  };

  return (
    <Grid item xs={2}>
      <FormControl fullWidth>
        <InputLabel>Currency</InputLabel>
        <Select
          data-testid="currency"
          value={value.toString()}
          label="Currency"
          onChange={handleChange}
        >
          {CURRENCIES.map((item) => (
            <MenuItem key={item.code} value={item.code}>
              {item.name}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </Grid>
  );
};
