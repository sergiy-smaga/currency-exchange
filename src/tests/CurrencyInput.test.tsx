import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { CurrencyInput } from '../components/CurrencyInput';
import CurrencyProvider from '../context/CurrencyContext';

const AllTheProviders = (props: { children: JSX.Element[] | JSX.Element }) => {
  return <CurrencyProvider>{props.children}</CurrencyProvider>;
};
function renderCurrencyInput() {
  return render(<CurrencyInput />, {
    wrapper: AllTheProviders,
  });
}

describe('CurrencyInput value', () => {
  it('snapshot tested', () => {
    const { asFragment } = renderCurrencyInput();
    expect(asFragment()).toMatchSnapshot();
  });
  // eslint-disable-next-line testing-library/no-render-in-setup
  beforeEach(() => renderCurrencyInput());

  it('render header', () => {
    const input = screen.getByTestId('currency-input');
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue(null);
  });

  it('change event give number value', () => {
    const input = screen.getByTestId('currency-input');
    userEvent.type(input, '123');
    expect(input).toHaveValue(123);
  });
});
