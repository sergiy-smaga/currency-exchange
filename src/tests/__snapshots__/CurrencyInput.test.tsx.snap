// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput value snapshot tested 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r1:"
        id=":r1:-label"
      >
        Change
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="currency-input"
          id=":r1:"
          type="number"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Change
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
